api/src/main/java/org/pmiops/workbench/exfiltration/EgressEventServiceImpl.java

```diff
@@ -100,8 +101,38 @@ public void handleEvent(SumologicEgressEvent event) {
     this.egressEventAuditor.fireEgressEventForUser(event, egressUser.get());
 
     Optional<DbEgressEvent> maybeEvent =
-        this.maybePersistEgressEvent(event, Optional.of(egressUser.get()), dbWorkspaceMaybe);
-    maybeEvent.ifPresent(e -> taskQueueService.pushEgressEventTask(e.getEgressEventId()));
+        this.maybePersistSumoEgressEvent(event, Optional.of(egressUser.get()), dbWorkspaceMaybe);
+    maybeEvent.ifPresent(e -> taskQueueService.pushEgressEventTask(e.getEgressEventId(), false));
+  }
+
+  @Override
+  public void handleVwbEvent(VwbEgressEventRequest egressEvent) {
+    Optional<DbUser> egressUser = userService.getByUsername(egressEvent.getUserEmail());
+    if (egressUser.isEmpty()) {
+      logger.severe(
+          String.format(
+              "An egress event happened for workspace: %s, but we're unable to find a user %s.",
+              egressEvent.getVwbWorkspaceId(), egressEvent.getUserEmail()));
+    }
+    this.egressEventAuditor.fireVwbEgressEvent(egressEvent, egressUser.get());
```

Comment:
line 111 implies that egressUser can be empty, line 117 makes a call to egressUser.get which might be empty.